### PR TITLE
[MM-49243] - Use paged requests to request the list of teams

### PR DESCRIPTION
**Pull request by EltonGohJH**
#### Summary
Use getPages when GetAllTeams is called with large sizes. Updated the test cases to match this new behaviour.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/25991
Jira https://mattermost.atlassian.net/browse/MM-49243

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/cmd/mmctl/commands/command.go [hunk 1]
```diff
@@ -188,7 +188,9 @@ func createCommandCmdF(c client.Client, cmd *cobra.Command, args []string) error
 func listCommandCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 	var teams []*model.Team
 	if len(args) < 1 {
-		teamList, _, err := c.GetAllTeams(context.TODO(), "", 0, 10000)
+		teamList, err := getPages(func(page, numPerPage int, etag string) ([]*model.Team, *model.Response, error) {
+			return c.GetAllTeams(context.TODO(), etag, page, numPerPage)
+		}, 250)
```
Comment (hanzei) [inline server/cmd/mmctl/commands/command.go:193]: I wonder if we should define a global const and use it everywhere instead of using a custom number.

Comment (hanzei) [inline server/cmd/mmctl/commands/command.go:193]: Other places in the code use 200 instead of 250. Let's stick with that number for consistency.

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the contribution :+1:

File: server/cmd/mmctl/commands/command.go (hunk 1 above)
Comment (EltonGohJH) [reply server/cmd/mmctl/commands/command.go:193]: I think that would be a great idea.

 I am still new to mattermost. Do you know in mattermost where do we generally store global const?
  I am thinking of storing that const `DefaultPageSize` in utils where the getPages function is located what do you think?

Comment (EltonGohJH) [reply server/cmd/mmctl/commands/command.go:193]: Sorry for oversight. Will fix that!

**Review by hanzei — APPROVED** [review]
LGTM :+1:

File: server/cmd/mmctl/commands/command.go (hunk 1 above)
Comment (hanzei) [reply server/cmd/mmctl/commands/command.go]: That sound good to me :+1:

**Comment by EltonGohJH** [comment]
@hanzei 
It seems that MySQL workflow is failing. Do I need to fix anything?

**Comment by EltonGohJH** [comment]
@BenCookie95
Can I check if this PR is ready to be merged? Do I need to do anything on my end?

**Comment by BenCookie95** [comment]
> @BenCookie95 Can I check if this PR is ready to be merged? Do I need to do anything on my end?

Nothing to do on your end right now, it will enter QA soon and if there are no issues we will merge it

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Regression tested MMCTL commands
- Used list for `team`, `command` and `webhook` 
- No issues found 

LGTM!

**Pull request merged**